src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -53,14 +53,14 @@ def getNewTimestamp(doc):
     Given a pileup doc - or a subset of it - return a dictionary
     with a couple timestamp attributes that need to be updated.
     :param doc: a python dictionary representing the pileup information
-    :return: a python dictionary to update the pileup object
+    :return: nothing
     """
     subDoc = {'lastUpdateTime': gmtimeSeconds()}
     if "active" in doc and doc['active'] is True:
         subDoc['activatedOn'] = gmtimeSeconds()
     elif "active" in doc and doc['active'] is False:
         subDoc['deactivatedOn'] = gmtimeSeconds()
-    return subDoc
+    doc.update(subDoc)
```

Comment:
Sorry Valentin, I didn't see it was actually returning a new dictionary. When I looked at it, I thought it was updating the same doc dictionary in place, hence my suggestion.
If you don't mind, I would say the previous code was looking better than this (and properly reflecting the getter method name).

Reply:
ok, I reverted the code